src/PrestaShopBundle/Controller/Admin/Sell/CustomerService/CustomerThreadController.php

```diff
@@ -359,6 +390,34 @@ public function bulkDeleteAction(Request $request): RedirectResponse
         return $this->redirectToRoute('admin_customer_threads');
     }
 
+    /**
+     * Update contact options
+     *
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     message="You do not have permission to update this.",
+     *     redirectRoute="admin_customer_threads_index"
+     * )
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function contactOptionsUpdate(Request $request): RedirectResponse
+    {
+        $this->getCommandBus()->handle(new UpdateContactOptionsCommand(
```

Comment:
Once you refacto your form like a configuration form, this controller should look more like this one https://github.com/PrestaShop/PrestaShop/blob/21f6c963dd152bdff0aaed9c2c5399df899d6f48/src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/CustomerPreferencesController.php#L76